src/text.h

```diff
@@ -344,6 +344,14 @@ struct Text {
         while (s.cursorend < (int)t.Len() && wxIsalnum(t[s.cursorend])) s.cursorend++;
     }
 
+    void SelectWordBefore(Selection &s) {
+        if (s.cursor <= 1) return;
+        s.cursorend = s.cursor--;
+        if (!wxIsalnum(t[s.cursor])) return;
+        while (s.cursor > 0 && wxIsalnum(t[s.cursor - 1])) s.cursor--;
+        while (s.cursorend < (int)t.Len() && wxIsalnum(t[s.cursorend])) s.cursorend++;

```

Comment:
These 3 lines are the same as the previous function.. pull them out into a helper function? or add a `bool before` arg? Whatever you think is nicer.

Reply:
Yeah, I more or less kept to the style of the surrounding code.
The other functions were also roughly the same so I duplicated the code here as well.

Reply:
Well, the style of the surrounding code is to not repeat yourself :)